layers.go

```diff
@@ -106,41 +107,35 @@ func (bd *bpLayersDir) findLayers(f func(layer bpLayer) bool) []bpLayer {
 	return selectedLayers
 }
 
-func (bd *bpLayersDir) newBPLayer(name string) *bpLayer {
+func (bd *bpLayersDir) newBPLayer(name, buildpackAPI string) *bpLayer {
```

Comment:
Seems like we should be eventually pushing this down into the `buildpack` package with the other buildpack API dependent logic.

Reply:
In order to do it, we need to make a big change since we need this information in the following flow (and in some other similar flows):
```
# exporter.go
addBuildpackLayers
    bpDir, err := readBuildpackLayersDir --> calls to newBPLayer
    bpDir.findLayers(forLaunch) --> calls to read --> calls to buildpack.DecodeLayerMetadataFile (needs the buildpack API version)
```

I'll keep it as is for now and we can make this change in a different PR.

Reply:
In order to prevent a big refactoring, I also had to add `Logger` to `newBPLayer` so it's now part of the `bpLayer` struct (see https://github.com/buildpacks/lifecycle/pull/537/commits/10e7898473c8b663ac444985d8b6fe429f57b8ba).
I added  a `TODO` comment so we'll change it in a different PR.